src/Symfony/Component/Yaml/Inline.php

```diff
@@ -669,6 +669,10 @@ private static function parseTag(string $value, int &$i, int $flags): ?string
         $nextOffset = $i + $tagLength + 1;
         $nextOffset += strspn($value, ' ', $nextOffset);
 
+        if ('' === $tag && (!isset($value[$nextOffset]) || \in_array($value[$nextOffset], [']', '}', ','], true))) {
```

Comment:
Are there legitimate use cases where `$tag` is the empty string, but we do not want to throw?

Reply:
Yes, `! scalar` (`! 12`, `! foo`). See https://github.com/symfony/symfony/blob/477e843790b240f20e9985f5804ed061ce534a74/src/Symfony/Component/Yaml/Tests/Fixtures/YtsSpecificationExamples.yml#L913.